Remove deprecation warnings for flags parameter

simp42 opened the pull request:
The flags parameter in preg_match and preg_match all is of type int (not string as previously asserted in the code) and not nullable. Passing null leads to a deprecation warning in newer PHP versions, which is now also checked for in the provided GitHub actions

simp42 committed:
adjusts flags parameter to be int

The fourth parameter of preg_match and preg_match_all is of type int and does not allow null values. With this change we now pass the correct default value if no other flags are given

gherkins commented:
thx a lot for that, 
but testing with multiple php versions is done via travis already.

strangely enough php8.1 seems to never have run yet, 
i guess changing only travis_ci.yml does not trigger build by itself.

If you wouldn't mind reducing to just the php file changes, I'll gladly accept and merge it! ✌️

simp42 commented:
I can do this and basically only keep c5b3d64 if you prefer. Shoudl I leave the changes to phpunit configuration intact that converts deprecation notices into failed tests?

gherkins commented:
> I can do this and basically only keep [c5b3d64](https://github.com/gherkins/regexpbuilderphp/commit/c5b3d64ce3755f47458031150220e967fdc30d43) if you prefer. Shoudl I leave the changes to phpunit configuration intact that converts deprecation notices into failed tests?

yes, that would be perfect - Thanks a lot!

simp42 committed:
updates phpunit configuration file

Updates configuration to follow current schema and tracks deprecations as exceptions

simp42 commented:
Ok, done. e763c13 updates the PHP code to remove the deprecated null parameter and e5c4535 updates phpunit configuration and makes it treat deprecations as failed tests

gherkins commented:
perfect, thanks a lot!

i just learned that travis isn't free for open source for quite some time now.. 🙈 
so github actions will actually be the way...

simp42 commented:
Ok. I can open a new pull request with the commit for the two actions I added again if you want

gherkins commented:
that would be supercool, sorry for the sloppy review ❤️
